File: WineriesApp/src/WineriesApp.UI/src/models/Filters/WineriesFilter.ts
import { BaseFilter } from "./BaseFilter";

export class WineriesFilter extends BaseFilter {
    public searchTerm: string | null = null;
    
    public ratings: number[] = [];

    public locations: string[] = [];

    constructor(searchTerm: string | null, ratings: number[], locations: string[], batchSize: number | null = null, batchIndex: number | null = null) {
        super(batchSize, batchIndex);
        this.searchTerm = searchTerm;
        this.ratings = ratings;
        this.locations = locations;
    }
}

File: WineriesApp/src/WineriesApp.Services/Models/Filters/WineriesFilter.cs
namespace WineriesApp.Services.Models.Filters
{
    public class WineriesFilter : BaseFilter
    {
        public string? SearchTerm { get; set; } = null;
    
        public double[] Ratings { get; set; } = Array.Empty<double>();

        public Guid[] Locations { get; set; } = Array.Empty<Guid>();
    }
}